WaveManager/src/main/java/com/github/sef24sp4/wavemanager/WaveManager.java

```diff
@@ -0,0 +1,201 @@
+package com.github.sef24sp4.wavemanager;
+
+import com.github.sef24sp4.common.enemy.CommonEnemy;
+import com.github.sef24sp4.common.enemy.EnemyRole;
+import com.github.sef24sp4.common.enemy.EnemySPI;
+import com.github.sef24sp4.common.entities.IEntity;
+import com.github.sef24sp4.common.interfaces.IGameSettings;
+import com.github.sef24sp4.common.vector.Coordinates;
+
+import java.util.*;
+
+public class WaveManager implements IWaveManager {
+
+	private Collection<IEntity> enemyComposition = new HashSet<>();
+
+	/**
+	 * used instead of a map.
+	 */
+	private final IGameSettings gameSettings;
+
+	private int waveNumber;
+
+	private long waveDifficulty;
+
+	private final int timeUntilNextWaveInSeconds;
```

Comment:
It might be relevant to rename to `waveCooldownSeconds` or `secondsBetweenWaves`

Reply:
I actually think the name in use is the most descriptive and can't be confused. I would like your reasoning on why you think that was a bad name, since this attribute describes exactly what that holds. The names with the "cooldown" might refer to when the wave can be used and the "seconds between" might assume that nothing else can happen between waves, and if there is implemeted something that happens between waves like a shop this attribute will describe something incorrect.